### PR TITLE
Add AlphaSOC Network Behavior Analytics integration

### PR DESCRIPTION
**Pull request by ioj**

### CODE REVIEW

File: Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+
+        return null;
+    }
+
+    var createAlertsURL = function (follow) {
+        var server = params.server.replace(/[\/]+$/, '');
+        return server + apiPathAlerts + "?follow=" + follow + "&threats=all"
+    }
+
+    var parseFollow = function () {
```
Comment (Itay4) [inline Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml]: Please declare function as an object.

File: Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+        var lastRun = getLastRun();
+        return lastRun && lastRun.follow ? lastRun.follow : "0";
+    }
+
+    var saveFollow = function (follow) {
```
Comment (Itay4) [inline Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml]: Please declare function as an object.

File: Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+        if (follow) {
+            setLastRun({ follow: follow });
+        }
+    }
+
+    var testGetAlerts = function () {
```
Comment (Itay4) [inline Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml]: Please declare function as an object.

File: Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+        var response = sendRequest(createAlertsURL("0"));
+        return response && response.alerts ? true : false;
+    }
+
+    var getAlerts = function (follow) {
```
Comment (Itay4) [inline Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml]: Please declare function as an object.

File: Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+        var response = sendRequest(createAlertsURL(follow));
+
+        return {
+            follow: response.follow,
+            content: response.alerts,
+            threats: response.threats
+        }
+    }
+
+    var appendIncidentsFromAlert = function (incidents, alert, threats) {
```
Comment (Itay4) [inline Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml]: Please declare function as an object.

File: Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+            if (!threat) {
+                logInfo('Error: Invalid alert content. Definition for threat "' + threatId + '" not found');
+                continue;
+            }
+
+            if (inScope(threat.severity, threat.policy)) {
+                alert.policy = threat.policy === true ? true : false;
+
+                incidents.push({
+                    "name": threat.title,
```
Comment (Itay4) [inline Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml]: Please change `"` to `'` - according to our coding conventions

File: Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+                    "occurred": occurredTs,
+                    "severity": convertSeverity(threat.severity),
+                    "labels": getLabels(alert),
+                    "rawJSON": JSON.stringify(alert)
+                });
+            }
+        }
+    }
+
+    var getOccurredTs = function (alert) {
```
Comment (Itay4) [inline Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml]: Please declare function as an object.

File: Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml [hunk 8]
```diff
@@ -0,0 +1,287 @@
...
+        try {
+            return alert.event.ts;
+        } catch (exc) {
+            return null;
+        }
+    }
+
+    var getLabels = function (alert) {
```
Comment (Itay4) [inline Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml]: Please declare function as an object.

File: Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml [hunk 9]
```diff
@@ -0,0 +1,287 @@
...
+        return labels;
+    }
+
+    var createLabelEntry = function (key, value) {
+        return { 'type': key, 'value': String(value) };
+    }
+
+    var inScope = function (severity, policy) {
+        var severityThereshold = getSeverityThereshold();
+        if (severity === null || severity === undefined || severity < severityThereshold) {
```
Comment (Itay4) [inline Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml]: Better to check for falsey condition by `!severity`

File: Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml [hunk 10]
```diff
@@ -0,0 +1,287 @@
...
+
+        var includePolicy = params.policy;
+        if (policy === true && !includePolicy) {
+            return false;
+        }
+
+        return true;
+    }
+
+    var convertSeverity = function (severity) {
```
Comment (Itay4) [inline Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml]: Please declare function as an object.

File: Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml [hunk 11]
```diff
@@ -0,0 +1,287 @@
...
+        } catch (exc) {
+            demistoSeverity = 0;
+        }
+
+        return demistoSeverity < 0 || demistoSeverity > 4 ? 0 : demistoSeverity;
+    }
+
+    var getSeverityThereshold = function () {
+        try {
+            var severity = parseInt(params.severity);
```
Comment (Itay4) [inline Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml]: Please handle parameters as constants at the beginning of the code the following.

File: Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml [hunk 12]
```diff
@@ -0,0 +1,287 @@
...
+    }
+
+    var createAlertsURL = function (follow) {
+        var server = params.server.replace(/[\/]+$/, '');
+        return server + apiPathAlerts + "?follow=" + follow + "&threats=all"
+    }
+
+    var parseFollow = function () {
+        var lastRun = getLastRun();
+        return lastRun && lastRun.follow ? lastRun.follow : "0";
```
Comment (Itay4) [inline Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml]: Please change `"` to `'` - according to our coding conventions

File: Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml [hunk 13]
```diff
@@ -0,0 +1,287 @@
...
+            var incidents = [];
+            if (alerts && alerts.content) {
+                for (var i = 0; i < alerts.content.length; i++) {
+                    appendIncidentsFromAlert(incidents, alerts.content[i], alerts.threats);
+                }
+            }
+
+            saveFollow(alerts.follow);
+            return JSON.stringify(incidents);
+        default:
```
Comment (Itay4) [inline Integrations/integration-AlphaSOC_Network_Behavior_Analytics.yml]: This is unnecessary, can be removed.

**Comment by anara123** [comment]
@Itay4 can we merge this pr?

**Pull request merged**